### PR TITLE
EMPs cause chameleon clothing to flicker

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Chameleon clothing produced by the syndicate has been found to
react negatively to EMPs, randomly switching forms for a time.
/:cl:

- ~~EMPing them when they're in the box is fine, but when they're being
worn, the game slows down a LOT, which I assume is the constant spamming
of the regenerate_icons() proc on the mob every time the chameleon
clothes change, which if they're wearing all 10 of them, that's a lot of
redundant calls.~~

Fixed it.

### CODE REVIEW

**Comment by XDTM** [comment]
It would probably save you the issue if it locked in a random disguise for a while, instead of flickering. Maybe even make clothes nodrop while theyre haywire.
It would make camo clothing stand out instead of basically screaming "this guy is wearing camo".

**Comment by optimumtact** [comment]
I'm sorry but what? You'll have to solve the issue you mentioned first, we're not going to merge something that can cause the game to slow down like that. Especially if it was noticeable with just one player on.

Investigate an overlay that follows the player using the effects system and make that an animation instead.

**Comment by coiax** [comment]
Oranges, DNM is in the title... I put this code up for help, not to get it merged.

**Comment by optimumtact** [comment]
oh, sorry I somehow missed that

**Comment by optimumtact** [comment]
I will leave this open for one more day, but it doesn't seem you are getting much help

File: code/modules/clothing/chameleon.dm [hunk 1]
```diff
@@ -149,6 +153,18 @@
 	select_look(owner)
 	return 1
 
+/datum/action/item_action/chameleon/change/proc/emp_randomise()
+	START_PROCESSING(SSprocessing, src)
```
Comment (Cyberboss) [inline code/modules/clothing/chameleon.dm]: Don't use SSprocessing, it just means you're trying to update basically every tick.

Use a timer set to go off every 3 seconds

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
@coiax Merge coiax/-tg-station#5

**Comment by coiax** [comment]
The issue isn't processing, the issue is that we're regenerating all icons 10 times every time it changes, because of the camo clothing taking the lazy option

**Comment by optimumtact** [comment]
cache as much as possible, use overlays for the flickering affect so you're only stripping out a single overlay at a time (i.e the icons below are unchanged)

**Comment by coiax** [comment]
@optimumtact REMOVE DNM, WIP AND STALE TAGS PLZ <3

**Pull request merged**